### PR TITLE
Update our inline observations to special case a few more intrinsics

### PR DESCRIPTION
**Pull request by tannergooding**
Specifically this handles the HWIntrinsic and Unsafe intrinsics that are "no-ops" or always produce a constant. It likewise adds some minimal handling for foldable Unsafe intrinsics involving constant inputs/args.

### CODE REVIEW

**Review by TIHan — COMMENTED** [review]
The changes look straight-forward. There are regressions though, are they expected and why do they occur?

**Review by EgorBo — APPROVED** [review]

File: src/coreclr/jit/fgbasic.cpp [hunk 1]
```diff
@@ -1236,6 +1249,300 @@ void Compiler::fgFindJumpTargets(const BYTE* codeAddr, IL_OFFSET codeSize, Fixed
...
+                            case NI_SRCS_UNSAFE_IsAddressLessThan:
+                            case NI_SRCS_UNSAFE_IsNullRef:
+                            case NI_SRCS_UNSAFE_Subtract:
+                            case NI_SRCS_UNSAFE_SubtractByteOffset:
+                            {
+                                // These are effectively primitive binary operations so the
+                                // handling roughly mirrors the handling for CEE_ADD and
+                                // friends that exists elsewhere in this method
+
+                                if (!preciseScan)
```
Comment (EgorBo) [inline src/coreclr/jit/fgbasic.cpp:1266]: I assume this path is added to improve the inliner used in R2R? I don't think we care much about it honestly. And NativeAOT for perf critical path uses new one (--Ot). All other places with `!preciseScan` were added for backward compatibility

**Comment by tannergooding** [comment]
Resolved merge conflict.

> There are regressions though, are they expected and why do they occur?

This adjusts the inlining observations by marking these APIs as having low cost and thus increases the chances that inlining can occur as compared to what it was previously.

We're effectively just handling the APIs that are true nops (actual zero cost) or which are being used in a way that lets us constant fold the expression.

File: src/coreclr/jit/fgbasic.cpp (hunk 1 above)
Comment (tannergooding) [reply src/coreclr/jit/fgbasic.cpp:1260]: Done for consistency with the other path. Figure it's better to match today and we can remove the handling altogether in a separate PR if it's no longer important/relevant.

**Comment by tannergooding** [comment]
@EgorBo, All just more inlining. Sometimes that results in dead code elimination, other times just larger code:

### jit-diff --pmi --frameworks
```
Found 302 files with textual diffs.

Summary of Code Size diffs:
(Lower is better)

Total bytes of base: 59570842
Total bytes of diff: 59569962
Total bytes of delta: -880 (-0.00 % of base)
Total relative delta: NaN
    diff is an improvement.
    relative diff is a regression.


Total byte diff includes -131 bytes from reconciling methods
        Base had    3 unique methods,      131 unique bytes
        Diff had    0 unique methods,        0 unique bytes

Top file regressions (bytes):
         861 : System.Management.dasm (0.22% of base)
         384 : System.Private.CoreLib.dasm (0.01% of base)
         299 : System.Private.Xml.dasm (0.01% of base)
         222 : System.Net.Security.dasm (0.13% of base)
         222 : System.Net.HttpListener.dasm (0.09% of base)
         211 : System.Security.Cryptography.dasm (0.02% of base)
         196 : System.Collections.Immutable.dasm (0.01% of base)
         103 : System.Data.Common.dasm (0.01% of base)
          88 : System.ObjectModel.dasm (0.14% of base)
          58 : System.Private.DataContractSerialization.dasm (0.01% of base)
           6 : Microsoft.CodeAnalysis.CSharp.dasm (0.00% of base)

Top file improvements (bytes):
        -819 : Microsoft.CodeAnalysis.VisualBasic.dasm (-0.01% of base)
        -568 : Microsoft.Diagnostics.Tracing.TraceEvent.dasm (-0.02% of base)
        -449 : Microsoft.CodeAnalysis.dasm (-0.02% of base)
        -328 : System.Threading.Tasks.Dataflow.dasm (-0.03% of base)
        -181 : System.Linq.Expressions.dasm (-0.02% of base)
        -151 : Microsoft.VisualBasic.Core.dasm (-0.03% of base)
        -144 : System.ComponentModel.Composition.Registration.dasm (-0.15% of base)
        -144 : System.Composition.Convention.dasm (-0.13% of base)
        -138 : xunit.runner.utility.netcoreapp10.dasm (-0.07% of base)
        -126 : System.Net.Http.dasm (-0.02% of base)
        -123 : System.IO.Packaging.dasm (-0.13% of base)
        -103 : System.Net.NameResolution.dasm (-0.21% of base)
         -82 : System.Numerics.Tensors.dasm (-0.03% of base)
         -66 : System.Linq.Parallel.dasm (-0.00% of base)
         -55 : System.Text.RegularExpressions.dasm (-0.01% of base)
         -48 : System.Text.Json.dasm (-0.00% of base)
          -5 : ILCompiler.Reflection.ReadyToRun.dasm (-0.00% of base)

28 total files with Code Size differences (17 improved, 11 regressed), 246 unchanged.

Top method regressions (bytes):
         391 (15.11% of base) : System.Management.dasm - System.Management.RelationshipQuery:ParseQuery(System.String):this
         262 ( 8.30% of base) : System.Management.dasm - System.Management.RelatedObjectQuery:ParseQuery(System.String):this
         211 (39.66% of base) : System.Security.Cryptography.dasm - Microsoft.Win32.SafeHandles.SafeCertContextHandleWithKeyContainerDeletion:DeleteKeyContainer(Microsoft.Win32.SafeHandles.SafeCertContextHandle)
         208 ( 8.33% of base) : System.Management.dasm - System.Management.SelectQuery:ParseQuery(System.String):this
         183 (125.34% of base) : System.Private.CoreLib.dasm - System.Threading.PortableThreadPool+CpuUtilizationReader:get_CurrentUtilization():int:this
         130 (24.25% of base) : System.Private.CoreLib.dasm - System.Diagnostics.Tracing.RuntimeEventSourceHelper:GetCpuUsage():double
         121 (201.67% of base) : System.Private.CoreLib.dasm - System.ComAwareWeakReference:SetTarget(byref,System.Object,System.ComAwareWeakReference+ComInfo)
         108 ( 4.56% of base) : System.Private.Xml.dasm - System.Xml.Serialization.XmlSerializationILGen:GenerateTypedSerializer(System.String,System.String,System.Xml.Serialization.XmlMapping,System.Xml.Serialization.CodeIdentifiers,System.String,System.String,System.String):System.String:this
         103 ( 8.63% of base) : System.Data.Common.dasm - System.Data.Common.ObjectStorage:Set(int,System.Object):this
          74 ( 1.21% of base) : Microsoft.Diagnostics.Tracing.TraceEvent.dasm - Microsoft.Diagnostics.Tracing.Parsers.ProviderManifest:ParseProviderEvents(System.Func`2[Microsoft.Diagnostics.Tracing.Parsers.DynamicTraceEventData,int],bool):this
          67 ( 1.61% of base) : Microsoft.CodeAnalysis.VisualBasic.dasm - Microsoft.CodeAnalysis.VisualBasic.LambdaRewriter:RewriteLambda(Microsoft.CodeAnalysis.VisualBasic.BoundLambda,Microsoft.CodeAnalysis.VisualBasic.Symbols.TypeSymbol,bool):Microsoft.CodeAnalysis.VisualBasic.BoundExpression:this
          49 ( 3.17% of base) : Microsoft.CodeAnalysis.VisualBasic.dasm - Microsoft.CodeAnalysis.VisualBasic.StateMachineRewriter`1[int]:CaptureParameterSymbol(Microsoft.CodeAnalysis.VisualBasic.Symbols.TypeSubstitution,Microsoft.CodeAnalysis.VisualBasic.Symbols.ParameterSymbol):int:this
          47 ( 3.03% of base) : Microsoft.CodeAnalysis.VisualBasic.dasm - Microsoft.CodeAnalysis.VisualBasic.StateMachineRewriter`1[ubyte]:CaptureParameterSymbol(Microsoft.CodeAnalysis.VisualBasic.Symbols.TypeSubstitution,Microsoft.CodeAnalysis.VisualBasic.Symbols.ParameterSymbol):ubyte:this
          46 ( 2.97% of base) : Microsoft.CodeAnalysis.VisualBasic.dasm - Microsoft.CodeAnalysis.VisualBasic.StateMachineRewriter`1[long]:CaptureParameterSymbol(Microsoft.CodeAnalysis.VisualBasic.Symbols.TypeSubstitution,Microsoft.CodeAnalysis.VisualBasic.Symbols.ParameterSymbol):long:this
          46 ( 2.96% of base) : Microsoft.CodeAnalysis.VisualBasic.dasm - Microsoft.CodeAnalysis.VisualBasic.StateMachineRewriter`1[short]:CaptureParameterSymbol(Microsoft.CodeAnalysis.VisualBasic.Symbols.TypeSubstitution,Microsoft.CodeAnalysis.VisualBasic.Symbols.ParameterSymbol):short:this
          46 ( 2.97% of base) : Microsoft.CodeAnalysis.VisualBasic.dasm - Microsoft.CodeAnalysis.VisualBasic.StateMachineRewriter`1[System.__Canon]:CaptureParameterSymbol(Microsoft.CodeAnalysis.VisualBasic.Symbols.TypeSubstitution,Microsoft.CodeAnalysis.VisualBasic.Symbols.ParameterSymbol):System.__Canon:this
          45 (17.11% of base) : System.Private.DataContractSerialization.dasm - System.Xml.XmlBinaryNodeWriter:WriteGuidArray(System.Guid[],int,int):this
          43 (116.22% of base) : System.Private.CoreLib.dasm - System.Runtime.CompilerServices.ConditionalWeakTable`2[System.__Canon,System.__Canon]:TryGetValue(System.__Canon,byref):bool:this
          37 (26.43% of base) : Microsoft.CodeAnalysis.dasm - Roslyn.Utilities.MultiDictionary`2[double,System.Nullable`1[int]]:Add(double,System.Nullable`1[int]):this
          37 (246.67% of base) : System.Net.Security.dasm - System.Collections.Generic.BidirectionalDictionary`2[double,System.Nullable`1[int]]:TryGetForward(double,byref):bool:this

Top method improvements (bytes):
         -91 (-6.27% of base) : Microsoft.CodeAnalysis.CSharp.dasm - Microsoft.CodeAnalysis.CSharp.ExpressionLambdaRewriter:VisitExpressionWithoutStackGuard(Microsoft.CodeAnalysis.CSharp.BoundExpression):Microsoft.CodeAnalysis.CSharp.BoundExpression:this
         -69 (-11.98% of base) : Microsoft.Diagnostics.Tracing.TraceEvent.dasm - Microsoft.Diagnostics.Tracing.Utilities.HistoryDictionary`2[System.__Canon,System.Nullable`1[int]]:Add(System.__Canon,long,System.Nullable`1[int],bool):this
         -52 (-100.00% of base) : System.Private.CoreLib.dasm - System.Collections.Generic.Dictionary`2[System.__Canon,System.__Canon]:TryGetValue(System.__Canon,byref):bool:this (1 base, 0 diff methods)
         -51 (-12.75% of base) : Microsoft.Diagnostics.Tracing.TraceEvent.dasm - Microsoft.Diagnostics.Tracing.Utilities.HistoryDictionary`2[int,System.Nullable`1[int]]:Add(int,long,System.Nullable`1[int],bool):this
         -51 (-12.53% of base) : Microsoft.Diagnostics.Tracing.TraceEvent.dasm - Microsoft.Diagnostics.Tracing.Utilities.HistoryDictionary`2[long,System.Nullable`1[int]]:Add(long,long,System.Nullable`1[int],bool):this
         -51 (-12.47% of base) : Microsoft.Diagnostics.Tracing.TraceEvent.dasm - Microsoft.Diagnostics.Tracing.Utilities.HistoryDictionary`2[short,System.Nullable`1[int]]:Add(short,long,System.Nullable`1[int],bool):this
         -51 (-12.66% of base) : Microsoft.Diagnostics.Tracing.TraceEvent.dasm - Microsoft.Diagnostics.Tracing.Utilities.HistoryDictionary`2[ubyte,System.Nullable`1[int]]:Add(ubyte,long,System.Nullable`1[int],bool):this
         -49 (-15.91% of base) : Microsoft.CodeAnalysis.VisualBasic.dasm - Microsoft.CodeAnalysis.VisualBasic.MethodToClassRewriter`1[double]:VisitLocal(Microsoft.CodeAnalysis.VisualBasic.BoundLocal):Microsoft.CodeAnalysis.VisualBasic.BoundNode:this
         -49 (-16.12% of base) : Microsoft.CodeAnalysis.VisualBasic.dasm - Microsoft.CodeAnalysis.VisualBasic.MethodToClassRewriter`1[long]:VisitLocal(Microsoft.CodeAnalysis.VisualBasic.BoundLocal):Microsoft.CodeAnalysis.VisualBasic.BoundNode:this
         -49 (-16.12% of base) : Microsoft.CodeAnalysis.VisualBasic.dasm - Microsoft.CodeAnalysis.VisualBasic.MethodToClassRewriter`1[System.__Canon]:VisitLocal(Microsoft.CodeAnalysis.VisualBasic.BoundLocal):Microsoft.CodeAnalysis.VisualBasic.BoundNode:this
         -48 (-14.29% of base) : Microsoft.CodeAnalysis.dasm - Microsoft.CodeAnalysis.CodeGen.ILBuilder:EmitFloat(float):this
         -48 (-15.84% of base) : Microsoft.CodeAnalysis.VisualBasic.dasm - Microsoft.CodeAnalysis.VisualBasic.MethodToClassRewriter`1[int]:VisitLocal(Microsoft.CodeAnalysis.VisualBasic.BoundLocal):Microsoft.CodeAnalysis.VisualBasic.BoundNode:this
         -48 (-15.79% of base) : Microsoft.CodeAnalysis.VisualBasic.dasm - Microsoft.CodeAnalysis.VisualBasic.MethodToClassRewriter`1[short]:VisitLocal(Microsoft.CodeAnalysis.VisualBasic.BoundLocal):Microsoft.CodeAnalysis.VisualBasic.BoundNode:this
         -48 (-0.73% of base) : System.Private.CoreLib.dasm - System.Diagnostics.Tracing.ManifestBuilder:CreateManifestString():System.String:this
         -47 (-15.51% of base) : Microsoft.CodeAnalysis.VisualBasic.dasm - Microsoft.CodeAnalysis.VisualBasic.MethodToClassRewriter`1[ubyte]:VisitLocal(Microsoft.CodeAnalysis.VisualBasic.BoundLocal):Microsoft.CodeAnalysis.VisualBasic.BoundNode:this
         -47 (-11.08% of base) : Microsoft.Diagnostics.Tracing.TraceEvent.dasm - Microsoft.Diagnostics.Tracing.Utilities.HistoryDictionary`2[double,System.Nullable`1[int]]:Add(double,long,System.Nullable`1[int],bool):this
         -44 (-15.66% of base) : Microsoft.CodeAnalysis.VisualBasic.dasm - Microsoft.CodeAnalysis.VisualBasic.MethodToClassRewriter`1[double]:VisitParameter(Microsoft.CodeAnalysis.VisualBasic.BoundParameter):Microsoft.CodeAnalysis.VisualBasic.BoundNode:this
         -43 (-13.69% of base) : Microsoft.CodeAnalysis.VisualBasic.dasm - Microsoft.CodeAnalysis.VisualBasic.MethodToClassRewriter`1[System.Nullable`1[int]]:VisitLocal(Microsoft.CodeAnalysis.VisualBasic.BoundLocal):Microsoft.CodeAnalysis.VisualBasic.BoundNode:this
         -42 (-15.27% of base) : Microsoft.CodeAnalysis.VisualBasic.dasm - Microsoft.CodeAnalysis.VisualBasic.MethodToClassRewriter`1[int]:VisitParameter(Microsoft.CodeAnalysis.VisualBasic.BoundParameter):Microsoft.CodeAnalysis.VisualBasic.BoundNode:this
         -42 (-15.27% of base) : Microsoft.CodeAnalysis.VisualBasic.dasm - Microsoft.CodeAnalysis.VisualBasic.MethodToClassRewriter`1[long]:VisitParameter(Microsoft.CodeAnalysis.VisualBasic.BoundParameter):Microsoft.CodeAnalysis.VisualBasic.BoundNode:this

Top method regressions (percentages):
          37 (308.33% of base) : System.Net.Security.dasm - System.Collections.Generic.BidirectionalDictionary`2[int,System.Nullable`1[int]]:TryGetForward(int,byref):bool:this
          37 (308.33% of base) : System.Net.HttpListener.dasm - System.Collections.Generic.BidirectionalDictionary`2[int,System.Nullable`1[int]]:TryGetForward(int,byref):bool:this
          37 (308.33% of base) : System.Net.Security.dasm - System.Collections.Generic.BidirectionalDictionary`2[long,System.Nullable`1[int]]:TryGetForward(long,byref):bool:this
          37 (308.33% of base) : System.Net.HttpListener.dasm - System.Collections.Generic.BidirectionalDictionary`2[long,System.Nullable`1[int]]:TryGetForward(long,byref):bool:this
          37 (308.33% of base) : System.Net.Security.dasm - System.Collections.Generic.BidirectionalDictionary`2[System.__Canon,System.Nullable`1[int]]:TryGetForward(System.__Canon,byref):bool:this
          37 (308.33% of base) : System.Net.HttpListener.dasm - System.Collections.Generic.BidirectionalDictionary`2[System.__Canon,System.Nullable`1[int]]:TryGetForward(System.__Canon,byref):bool:this
          34 (283.33% of base) : Microsoft.CodeAnalysis.dasm - Microsoft.Cci.MetadataWriter+HeapOrReferenceIndex`1[int]:TryGetValue(int,byref):bool:this
          34 (283.33% of base) : Microsoft.CodeAnalysis.dasm - Microsoft.Cci.MetadataWriter+HeapOrReferenceIndex`1[long]:TryGetValue(long,byref):bool:this
          34 (283.33% of base) : Microsoft.CodeAnalysis.dasm - Microsoft.Cci.MetadataWriter+HeapOrReferenceIndex`1[System.__Canon]:TryGetValue(System.__Canon,byref):bool:this
          37 (246.67% of base) : System.Net.Security.dasm - System.Collections.Generic.BidirectionalDictionary`2[double,System.Nullable`1[int]]:TryGetForward(double,byref):bool:this
          37 (246.67% of base) : System.Net.HttpListener.dasm - System.Collections.Generic.BidirectionalDictionary`2[double,System.Nullable`1[int]]:TryGetForward(double,byref):bool:this
          37 (246.67% of base) : System.Net.Security.dasm - System.Collections.Generic.BidirectionalDictionary`2[ubyte,System.Nullable`1[int]]:TryGetForward(ubyte,byref):bool:this
          37 (246.67% of base) : System.Net.HttpListener.dasm - System.Collections.Generic.BidirectionalDictionary`2[ubyte,System.Nullable`1[int]]:TryGetForward(ubyte,byref):bool:this
          37 (231.25% of base) : System.Net.Security.dasm - System.Collections.Generic.BidirectionalDictionary`2[short,System.Nullable`1[int]]:TryGetForward(short,byref):bool:this
          37 (231.25% of base) : System.Net.HttpListener.dasm - System.Collections.Generic.BidirectionalDictionary`2[short,System.Nullable`1[int]]:TryGetForward(short,byref):bool:this
          34 (226.67% of base) : Microsoft.CodeAnalysis.dasm - Microsoft.Cci.MetadataWriter+HeapOrReferenceIndex`1[double]:TryGetValue(double,byref):bool:this
          34 (226.67% of base) : Microsoft.CodeAnalysis.dasm - Microsoft.Cci.MetadataWriter+HeapOrReferenceIndex`1[ubyte]:TryGetValue(ubyte,byref):bool:this
          34 (212.50% of base) : Microsoft.CodeAnalysis.dasm - Microsoft.Cci.MetadataWriter+HeapOrReferenceIndex`1[short]:TryGetValue(short,byref):bool:this
         121 (201.67% of base) : System.Private.CoreLib.dasm - System.ComAwareWeakReference:SetTarget(byref,System.Object,System.ComAwareWeakReference+ComInfo)
         183 (125.34% of base) : System.Private.CoreLib.dasm - System.Threading.PortableThreadPool+CpuUtilizationReader:get_CurrentUtilization():int:this

Top method improvements (percentages):
         -38 (-100.00% of base) : System.Private.CoreLib.dasm - System.Collections.Generic.Dictionary`2[int,System.__Canon]:get_Item(int):System.__Canon:this (1 base, 0 diff methods)
         -52 (-100.00% of base) : System.Private.CoreLib.dasm - System.Collections.Generic.Dictionary`2[System.__Canon,System.__Canon]:TryGetValue(System.__Canon,byref):bool:this (1 base, 0 diff methods)
         -41 (-100.00% of base) : System.Private.CoreLib.dasm - System.Collections.Generic.Dictionary`2[ulong,System.__Canon]:get_Item(ulong):System.__Canon:this (1 base, 0 diff methods)
         -15 (-28.30% of base) : System.Numerics.Tensors.dasm - System.Numerics.Tensors.SparseTensor`1[long]:GetValue(int):long:this
         -16 (-27.59% of base) : System.Numerics.Tensors.dasm - System.Numerics.Tensors.SparseTensor`1[double]:GetValue(int):double:this
         -14 (-27.45% of base) : System.Numerics.Tensors.dasm - System.Numerics.Tensors.SparseTensor`1[int]:GetValue(int):int:this
         -29 (-27.10% of base) : Microsoft.Diagnostics.Tracing.TraceEvent.dasm - Microsoft.Diagnostics.Tracing.Utilities.HistoryDictionary`2[int,System.Nullable`1[int]]:Remove(int):this
         -14 (-26.92% of base) : System.Numerics.Tensors.dasm - System.Numerics.Tensors.SparseTensor`1[ubyte]:GetValue(int):ubyte:this
         -13 (-26.53% of base) : Microsoft.CodeAnalysis.dasm - Roslyn.Utilities.MultiDictionary`2[int,System.Nullable`1[int]]:get_Item(int):Roslyn.Utilities.MultiDictionary`2+ValueSet[int,System.Nullable`1[int]]:this
         -13 (-26.53% of base) : Microsoft.CodeAnalysis.dasm - Roslyn.Utilities.MultiDictionary`2[long,System.Nullable`1[int]]:get_Item(long):Roslyn.Utilities.MultiDictionary`2+ValueSet[long,System.Nullable`1[int]]:this
         -13 (-26.53% of base) : Microsoft.CodeAnalysis.dasm - Roslyn.Utilities.MultiDictionary`2[System.__Canon,System.Nullable`1[int]]:get_Item(System.__Canon):Roslyn.Utilities.MultiDictionary`2+ValueSet[System.__Canon,System.Nullable`1[int]]:this
         -18 (-26.47% of base) : System.ComponentModel.Composition.Registration.dasm - System.ComponentModel.Composition.Registration.PartBuilder`1+ConstructorExpressionAdapter[double]:ConfigureConstructorImports(System.Reflection.ParameterInfo,System.ComponentModel.Composition.Registration.ImportBuilder):this
         -18 (-26.47% of base) : System.ComponentModel.Composition.Registration.dasm - System.ComponentModel.Composition.Registration.PartBuilder`1+ConstructorExpressionAdapter[int]:ConfigureConstructorImports(System.Reflection.ParameterInfo,System.ComponentModel.Composition.Registration.ImportBuilder):this
         -18 (-26.47% of base) : System.ComponentModel.Composition.Registration.dasm - System.ComponentModel.Composition.Registration.PartBuilder`1+ConstructorExpressionAdapter[long]:ConfigureConstructorImports(System.Reflection.ParameterInfo,System.ComponentModel.Composition.Registration.ImportBuilder):this
         -18 (-26.47% of base) : System.ComponentModel.Composition.Registration.dasm - System.ComponentModel.Composition.Registration.PartBuilder`1+ConstructorExpressionAdapter[short]:ConfigureConstructorImports(System.Reflection.ParameterInfo,System.ComponentModel.Composition.Registration.ImportBuilder):this
         -18 (-26.47% of base) : System.ComponentModel.Composition.Registration.dasm - System.ComponentModel.Composition.Registration.PartBuilder`1+ConstructorExpressionAdapter[System.__Canon]:ConfigureConstructorImports(System.Reflection.ParameterInfo,System.ComponentModel.Composition.Registration.ImportBuilder):this
         -18 (-26.47% of base) : System.ComponentModel.Composition.Registration.dasm - System.ComponentModel.Composition.Registration.PartBuilder`1+ConstructorExpressionAdapter[System.Nullable`1[int]]:ConfigureConstructorImports(System.Reflection.ParameterInfo,System.ComponentModel.Composition.Registration.ImportBuilder):this
         -18 (-26.47% of base) : System.ComponentModel.Composition.Registration.dasm - System.ComponentModel.Composition.Registration.PartBuilder`1+ConstructorExpressionAdapter[System.Numerics.Vector`1[float]]:ConfigureConstructorImports(System.Reflection.ParameterInfo,System.ComponentModel.Composition.Registration.ImportBuilder):this
         -18 (-26.47% of base) : System.ComponentModel.Composition.Registration.dasm - System.ComponentModel.Composition.Registration.PartBuilder`1+ConstructorExpressionAdapter[ubyte]:ConfigureConstructorImports(System.Reflection.ParameterInfo,System.ComponentModel.Composition.Registration.ImportBuilder):this
         -18 (-26.47% of base) : System.Composition.Convention.dasm - System.Composition.Convention.PartConventionBuilder`1+ConstructorExpressionAdapter[double]:ConfigureConstructorImports(System.Reflection.ParameterInfo,System.Composition.Convention.ImportConventionBuilder):this

411 total methods with Code Size differences (272 improved, 139 regressed), 375446 unchanged.
```

**Comment by EgorBo** [comment]
Oa, nice to see cases where extra inlining leads to a negative diff 🙂

**Comment by tannergooding** [comment]
No diffs for benchmarks, which makes sense given many of them are small 1-line methods:

### jit-diff --pmi --benchmarks
```
Found 105 files with textual diffs.

Summary of Code Size diffs:
(Lower is better)

Total bytes of base: 502878
Total bytes of diff: 502878
Total bytes of delta: 0 (0.00 % of base)


0 total files with Code Size differences (0 improved, 0 regressed), 98 unchanged.

0 total methods with Code Size differences (0 improved, 0 regressed), 2797 unchanged.
```

**Pull request merged**